### PR TITLE
Feature/restructure and rename tables

### PR DESCRIPTION
**Pull request by tinneuro**
Preparations done due to sharing most important functions for integration into local systems.
- Aggregation of linelist is now a public function, not ultimately necessary as aggregation is done inside the get_signals() but I think good to have.

Work on the results_table.R script:
- seperating the formating and preprocessing of tables such that users can also only retrieve the preprocessed tables using our functions
- renaming the functions to improve clarity (hopefully)
- adding updated and new examples to the function description and updating the description
- change of the used table functions in the report and signals tab
I am also happy if someone disagrees with naming or has another suggestion. I just gave it a try to make it more clear when I was seperating the functions anyway.

### CODE REVIEW

**Comment by tinneuro** [comment]
Hi @tomasmartbert ,

can you manage to do the review today?
Thanks!

**Comment by tomasmartbert** [comment]
Nice work! Here a few preliminary comments - I'll get back with some suggestions later.

> Preparations done due to sharing most important functions for integration into local systems.
> 
> * Aggregation of linelist is now a public function, not ultimately necessary as aggregation is done inside the get_signals() but I think good to have.
Should we consider exposing only the minimal set of functions, hence not export aggregation function?
> 
> Work on the results_table.R script:
> 
> * seperating the formating and preprocessing of tables such that users can also only retrieve the preprocessed tables using our functions
> * renaming the functions to improve clarity (hopefully)
I feel the names are little too long and could be made more precise - but otherwise a nice refactoring.
> * adding updated and new examples to the function description and updating the description
> * change of the used table functions in the report and signals tab
>   I am also happy if someone disagrees with naming or has another suggestion. I just gave it a try to make it more clear when I was seperating the functions anyway.

File: R/tool_functions.R [hunk 1]
```diff
@@ -594,8 +594,7 @@ get_signals_stratified <- function(data,
 #'
 #' @examples
 #' \dontrun{
-#' data <- read.csv("data/input/input.csv")
-#' results <- get_signals(data,
+#' results <- get_signals(preprocess_data(input_example),
```
Comment (tomasmartbert) [inline R/tool_functions.R:597]: Use %>% pipe would make the dataflow clearer, I think.

**Review by tomasmartbert — COMMENTED** [review]
I have to think a bit more on this and do some testing. Here are some preliminary thoughts, though. 
Perhaps it would make sense to control the output type through options to a single make_signals_table() instead of a splitup of the signal table function?

**Comment by tinneuro** [comment]
> I have to think a bit more on this and do some testing. Here are some preliminary thoughts, though.
> Perhaps it would make sense to control the output type through options to a single make_signals_table() instead of a splitup of the signal table function?

That is a good point. I am also thinking about it and can maybe implement an example alternative.

> I feel the names are little too long and could be made more precise - but otherwise a nice refactoring.
Yeah I agree. You have any suggestions? But maybe first I try the above mentioned improvement.

**Comment by tinneuro** [comment]
@tomasmartbert Are you back today to review? I want to get this merged today for the new version.

**Comment by tomasmartbert** [comment]
> @tomasmartbert Are you back today to review? I want to get this merged today for the new version.

Hi @tinneuro I don't have much time the rest of this week - I'll see if I can make a quick look today.

**Comment by tinneuro** [comment]
Thanks, If not no problem maybe @alchalu or @aj-thl can review otherwise

File: R/results_table.R [hunk 2]
```diff
@@ -202,48 +219,56 @@ create_results_table <- function(data,
     data <- data %>% dplyr::filter(alarms)
   }
 
+  return(data)
```
[the remark this answers is not in the document]
Comment (alchalu) [reply R/results_table.R:222]: Yeah we still have a couple insances where we use `return()`. I am not sure if we should remove them all at once or whereever it pops up (like here)

**Comment by tomasmartbert** [comment]
> > Should we consider exposing only the minimal set of functions, hence not export aggregation function?
> 
> Yes let's start with your suggestion. We can also discuss this in the next tool dev meeting and we can see whether some countries who want to integrate it into their systems need more public functions.
Maybe it would be a good thing to discuss at the meeting - it is maybe of more conceptual character.
> 
> > I have to think a bit more on this and do some testing. Here are some preliminary thoughts, though.
> > Perhaps it would make sense to control the output type through options to a single make_signals_table() instead of a splitup of the signal table function?
> 
> I also thought a bit and my first intuition is that I would tend to keep them as two seperate functions as I think it improves testability of each individual one and maybe also readability of the code. For the user to just have one function might be nice I agree. But if you have good arguments to put it in one happy to do that. I will work on the naming now.

That's a good point regarding testability, I think. I'm not so familiar with these functions so I'll yet have to dig into the code to give some more thorough feedback. From the user's point of view a simple wrapper function to be exposed might work to not be overwhelmed be the sheer number of functions. E.g. something in line with `create_signals_table(data, options, format = c("data.frame", "gt", "DataTable"))`

File: R/results_table.R [hunk 3]
```diff
@@ -31,13 +31,14 @@ get_float_columns <- function(data) {
...
+#' Format the signal results to an interactive or static table
 #'
 #' This function creates an interactive DataTable or a static gt table depending
-#' on the value of the `interactive` parameter. It can also format float columns
-#' to have two decimal places.
+#' on the value of the `interactive` parameter. It takes the stratum and category columns and transforms those to be headers of sections containing all signal results for these strata. It colors rows red for tables containing the any_alarms column when there was >= 1 alarm. It makes the table downloadable as CSV, Excel, PDF.
 #'
 #' @param data A data frame.
+#'
+#' @param positive_only Logical indicating whether to filter only those signal results where a signal was generated (default is TRUE). When TRUE then the signals column is not removed from the table.
```
Comment (alchalu) [inline R/results_table.R:41]: The description of positive_only is not super clear to me, Maybe something like: Logical indicating whether to filter the signal results to include only the weeks when a signal was generated (default is TRUE). If set to TRUE, the signals column will not be removed from the table.

**Comment by alchalu** [comment]
I've noticed that I'm struggling a bit myslef to use the various functions outside of the app, especially after not looking at the project for a while. I think we still have room to improve with userfriendliness of everything that's exported outside the app. I agree this is more of a conceptual issue, and I'm not sure if this PR is the right place to discuss it. 

But maybe for the next meeting it would be a good step to discuss and define all the functions that we want to make available for the user.

**Comment by tinneuro** [comment]
@tomasmartbert thanks for your suggestion `make_signals_table(data, options, format = c("data.frame", "gt", "DataTable"))` this was good. I changed my build_signals_table() now so it can take an argument like format and I think and hope it is nice and intuitive to use now

**Pull request merged**